testsuites/syncgateway/functional/tests/test_cbs_collections.py

```diff
@@ -326,6 +328,93 @@ def test_restricted_collection(scopes_collections_tests_fixture):
     assert(doc_3_key not in all_docs_ids), "Sync Gateway contains document from server restricted collection. Document ID " + doc_3_key + "under " + bucket + "." + scope + "." + third_collection
 
 
+@pytest.mark.syncgateway
+@pytest.mark.collections
+def test_user_collections_access(scopes_collections_tests_fixture):
+    """
+    1. Add second collection to SGW db and server, using sync functions to restrict collection/scope level access
+    2. Create two users, one with scope level access, one with collection level access
+    3. Check document upload is restricted to correct access level
+    4. Check document retrieval via all_docs and document ID is restricted to correct access level
+    5. Check document deletion is restricted to correct access level
+    """
+
+    if is_using_views:
+        pytest.skip("""It is not necessary to run scopes and collections tests with views.
+                When it is enabled, there is a problem that affects the rest of the tests suite.""")
+
+    sg_client, sg_admin_url, db, scope, collection = scopes_collections_tests_fixture
+    random_suffix = str(uuid.uuid4())[:8]
+
+    # 1. Add second collection to SGW db and server, using sync functions and collection_access fields to restrict collection/scope level access
+    second_collection = "second_collection" + random_suffix
+    cb_server.create_collection(bucket, scope, second_collection)
+
+    test_sync = f"function(doc, oldDoc) {{requireAccess(\"{collection}\"); channel(\"{collection}\")}}"
+    test_sync_2 = f"function(doc, oldDoc) {{requireAccess(\"{second_collection}\"); channel(\"{second_collection}\")}}"
+
+    db_config = {"bucket": bucket, "scopes": {scope: {"collections": {collection: {"sync": test_sync}, second_collection: {"sync": test_sync_2}}}}, "num_index_replicas": 0}
+    admin_client.post_db_config(db, db_config)
+    admin_client.wait_for_db_online(db, 60)
+
+    # 2. Create two users, one with scope level access, one with collection level access
+    scope_user = "scope_user" + random_suffix
+    collection_user = "collection_user" + random_suffix
+
+    scope_user_access = {scope: {collection: {"admin_channels": [collection]}, second_collection: {"admin_channels": [second_collection]}}}
```

Comment:
Isn't the scope name missing here?

Reply:
No, `scope` here is the string variable for the scope name as yielded by `scopes_collections_tests_fixture`